andela/bp-esa-frontend
# Configure Eslint

#### What does this PR do?
- Configure ESLINT using the Airbnb style guide.
- Ejects the Create-React-App configurations.
- Refactored some of the react codes that do not obey the with the Airbnb style guide.
- Changed all instances of `React.Component` to `React.PureComponent` for optimization sake
#### Description of Task to be completed?
Configure ESLINT using the Airbnb style guide.
#### How should this be manually tested?

- Clone and set up the project using the instruction in the README.
- The ESLINT configurations can be found in `.eslintrc.json` in the project's root directory
- Run `$ npm run lint` to lint the react codes
- Any code going against the Airbnb style guide should be flagged in the console 

#### Any background context you want to provide?
Create-React-App has its own(abstracted) eslint configuration which makes use of the `react-app` style guide. I had to eject Create-React-App configurations using `$ npm run eject` to be able to change the style guide. Consequently, most of the file changes in this Pull Request are there as a reuslt of the ejected configuration.
#### What are the relevant pivotal tracker stories?
This pull request is for a task under the story #160437219
